### ISSUE TITLE
Use hash for admin password for oci_database_autonomous_database resource state

### ISSUE THREAD

**Issue by anilarora** [terraform-providers/terraform-provider-oci#847]
### Terraform Version and Provider Version

terraform 0.11
terraform-provider version 3.37

### Affected Resource(s)

oci_database_autonomous_database

### Expected Behavior

Password should not be in plaintext in statefile.

### Actual Behavior

In the state file, I see something like

"oci_database_autonomous_database.autonomous_database": {
	"type": "oci_database_autonomous_database",
	"depends_on": [],
	"primary": {
		"id": "<>",
		"attributes": {
			"admin_password": "BEstr0ng_#12",
		}
}
...truncated...

### Steps to Reproduce

Create a resource of the type oci_database_autonomous_database and provide a password.  Run terraform apply.

**Comment by afedorch** [comment]
is related to https://github.com/hashicorp/terraform/issues/516

**Comment by anilarora** [comment]
Could something like https://github.com/terraform-providers/terraform-provider-tls/blob/master/tls/resource_locally_signed_cert.go be done so that only the hash is stored in the statefile?

**Comment by afedorch** [comment]
https://github.com/hashicorp/terraform/issues/21758

**Comment by anilarora** [comment]
FYI. I don’t believe that this particular issue is blocked by the mentioned issue because the incoming password is technically not computed in every case. So you can still successfully apply the state function for the inputs.

**Comment by afedorch** [comment]
We will check if there is a way to address sensitive information in the state file only for the `oci_database_autonomous_database` resource. 
Currently, the common approach is treating the whole state as a sensitive artifact. Please see https://github.com/hashicorp/terraform/issues/21758 for more details. Thanks!

**Comment by anilarora** [comment]
This should also be applied to the `oci_database_db_home` resource as well.

**Comment by ravinitp** [comment]
We are very sorry that we couldn't respond to each and every issue reported on GitHub. 
Although we have refined the process to prioritize customer issues on GitHub, since this issue was reported a while ago, there is a good chance it may have been fixed in the latest version of Terraform Provider OCI. 
If you are still experiencing this issue, please create a new issue and label it as Bug.
